backend/molgenis-emx2-semantics/src/main/java/org/molgenis/emx2/beaconv2/endpoints/individuals/Diseases.java

```diff
@@ -5,25 +5,38 @@
 import com.fasterxml.jackson.annotation.JsonAutoDetect;
 import java.util.List;
 import java.util.Map;
+import org.molgenis.emx2.beaconv2.common.AgeAndAgeGroup;
 import org.molgenis.emx2.beaconv2.common.OntologyTerm;
+import org.molgenis.emx2.utils.TypeUtils;
 
 @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
 public class Diseases {
 
   private OntologyTerm diseaseCode;
-  private AgeOfOnset ageOfOnset;
+  private AgeAndAgeGroup ageOfOnset;
+  private AgeAndAgeGroup ageAtDiagnosis;
   private Boolean familyHistory;
   private OntologyTerm severity;
   private OntologyTerm stage;
 
   public static Diseases[] get(Object diseases) {
+    if (diseases == null) {
+      return null;
+    }
     List<Map<String, Object>> diseasesCast = (List<Map<String, Object>>) diseases;
     Diseases[] result = new Diseases[diseasesCast.size()];
     for (int i = 0; i < diseasesCast.size(); i++) {
       Map map = diseasesCast.get(i);
       Diseases d = new Diseases();
       d.diseaseCode = mapToOntologyTerm((Map) map.get("diseaseCode"));
```

Comment:
Might be good to have all theses strings in a constants file, so if they change they change in only 1 place.

Reply:
makes sense. done.